tools/audio/extraction/audiobank_structs.py

```diff
@@ -0,0 +1,406 @@
+# audiobank_structs.py
+# SPDX-FileCopyrightText: © 2024 ZeldaRET
+# SPDX-License-Identifier: CC0-1.0
+#
+#   This file implements reading various structures resident to the Audiobank files.
+#   Additionally handles:
+#    - Linking with finalized samples
+#    - Writing xml elements representing these structures in soundfont xmls
+#
+
+import struct
+from enum import IntEnum
+
+from audio_tables import AudioStorageMedium
+from tuning import rate_from_tuning, pitch_names
+from util import XMLWriter
+
+VADPCM_VERSTAMP = 1
+
+class AudioSampleCodec(IntEnum):
+    CODEC_ADPCM = 0
+    CODEC_S8 = 1
+    CODEC_S16_INMEMORY = 2
+    CODEC_SMALL_ADPCM = 3
+    CODEC_REVERB = 4
+    CODEC_S16 = 5
+
+
+
+class SoundFontSample: # SampleHeader ?
+    """
+    typedef struct {
+        /* 0x00 */ u32 codec : 4;
+        /* 0x00 */ u32 medium : 2;      // storage medium determines which of the two sample bank ids to use when relocating sampleAddr
+        /* 0x00 */ u32 cached : 1;
+        /* 0x00 */ u32 isRelocated : 1;
+        /* 0x01 */ u32 size : 24;
+        /* 0x04 */ u8* sampleAddr;      // offset into the sample bank associated with this soundfont
+        /* 0x08 */ AdpcmLoop* loop;
+        /* 0x0C */ AdpcmBook* book;
+    } SoundFontSample; // size = 0x10
+    """
+    SIZE = 0x10
+
+    def __init__(self, data):
+        bits, self.sample_addr, self.loop, self.book = struct.unpack(">IIII", data[:0x10])
+
+        self.codec        = AudioSampleCodec((bits >> 28) & 0b1111)
+        self.medium       = AudioStorageMedium((bits >> 26) & 0b11)
+        self.cached       = bool((bits >> 25) & 1)
+        self.is_relocated = bool((bits >> 24) & 1)
+        self.size         = (bits >> 0) & 0b111111111111111111111111
+
+        assert self.book != 0
+        assert self.loop != 0
+        assert self.codec in [AudioSampleCodec.CODEC_ADPCM, AudioSampleCodec.CODEC_SMALL_ADPCM]
+        assert self.medium == 0
+        assert not self.is_relocated # Not relocated in ROM
+
+    def to_xml(self, xml : XMLWriter, name : str, rate_override = None, note_override = None):
+        # <Sample Name="SAMPLE_NAME" SampleRate="32000" BaseNote="C4" IsDD="false" Cached="false">
```

Comment:
I'm assuming this is meant to be an example showing what is written? If so could you add a comment saying that.